go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -212,7 +223,7 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 	bsnc.logicalPortCache.remove(pod, nadName)
 
 	// TBD namespaceInfo needed when multi-network policy support is added
-	pInfo, err := bsnc.deletePodLogicalPort(pod, portInfo, nadName)
+	pInfo, err := bsnc.deletePodLogicalPort(pod, portInfo, nadName, bsnc)
```

Comment:
I found there are a few functions like this, the same _bsnc_ is both the function receiver and the argument, which is a bit confusing to me. May I ask the reason for this?

Reply:
Sure, so all the functions that have such argument, are actually embedded `BaseNetworkController` functions, so the call is equivalent to `bsnc.BaseNetworkController.deletePodLogicalPort(pod, portInfo, nadName, bsnc)`.
Then in the default network controller part, there is a `oc.deletePodLogicalPort(pod, portInfo, ovntypes.DefaultNetworkName, oc)`, which is also just `oc.BaseNetworkController.deletePodLogicalPort(pod, portInfo, ovntypes.DefaultNetworkName, oc)`.

We can redefine these functions like
```
func (bsnc *BaseSecondaryNetworkController) deletePodLogicalPort(pod *kapi.Pod, portInfo *lpInfo,
	nadName string) (*lpInfo, error) {
   return bsnc.BaseNetworkController.deletePodLogicalPort(pod, portInfo, nadName, bsnc)
}
```
to make the call itself as it was, but that will add some boilerplate for every controller, so I wasn't sure if we want to do this.
I can also replace the call with `bsnc.BaseNetworkController.deletePodLogicalPort(pod, portInfo, nadName, bsnc)` it that will make things more obvious

Reply:
I see. I still think this change can be confusing. That said, the code looks correct. I'll leave @trozet @girishmg to decide if the change is worth it. Thanks!